src/commontransformations.jl

```diff
@@ -0,0 +1,352 @@
+    xz = q.v1 * q.v3
+    yw = q.v2 * q.s
+    yz = q.v2 * q.v3
+    xw = q.s  * q.v1
+
+    # initialize rotation part
+    return @fsa([1 - 2 * (yy + zz)    2 * (xy - zw)       2 * (xz + yw);
+                 2 * (xy + zw)        1 - 2 * (xx + zz)   2 * (yz - xw);
+                 2 * (xz - yw)        2 * (yz + xw)       1 - 2 * (xx + yy)])
+
+    # This leads to these derivative matrices:
+
+    dR/ds =
+    [ 0    -2*v3    2*v2;
+     2*v3     0    -2*v1;
+    -2*v2  2*v1     0   ]
+
+    dR/dv1 =
+    [ 0  2*v2  2*v3 ;
+     2*v2 -4*v1 -2*s;
+     2*v3  2*s  -4*v1]
+
+    dR/dv2 =
+    [-4*v2  2*v1  2*s;
+      2*v1  0     2*v3;
+     -2*s   2*v3  -4*v2]
+
+    dR/dv3 =
+    [-4*v3  -2*s  2*v1;
+     2*s    -4*v3  2*v2;
+     2*v1   2*v2   0 ]
+    =#
+    s = trans.rotation.s
+    v1 = trans.rotation.v1
+    v2 = trans.rotation.v2
+    v3 = trans.rotation.v3
+
+    @fsa [ 2(-v3*x[2]+v2*x[3])  2(v2*x[2]+v3*x[3])            2(-2*v2*x[1]+v1*x[2]+s*x[3])  2(-2*v3*x[1]-s*x[2]+v1*x[3]) ;
+           2(v3*x[1]-v1*x[3])   2(v2*x[1]-2*v1*x[2]- s*x[3])  2(v1*x[1]+v3*x[3])            2(s*x[1]-2*v3*x[2]+v2*x[3])  ;
+           2(-v2*x[1]+v1*x[2])  2(v3*x[1]+ s*x[2]-2*v1*x[3])  2(-s*x[1]+v3*x[2]-2*v2*x[3])  2(v1*x[1]+v2*x[2])           ]
+end
+
+#function transform_deriv_params{T1,T2}(trans::Rotation{EulerAngles{T1},T1}, x::FixedVector{3,T2})
+#end
+
+Base.inv(trans::Rotation) = Rotation(nothing, trans.matrix')
+Base.inv{T <: Union{Quaternion, SpQuat}}(trans::Rotation{T}) = Rotation(inv(trans.rotation), trans.matrix')
+
+compose(t1::Rotation, t2::Rotation) = Rotation(nothing, t1.matrix*t2.matrix) # A bit lossy for transform_deriv_params, but otherwise probably the most efficient choice
+
+
+#############################
+# Individual axis rotations #
+#############################
+immutable RotationXY{T} <: AbstractTransformation{FixedVector{3}, FixedVector{3}}
+    angle::T
+    sin::T
+    cos::T
+end
+immutable RotationYZ{T} <: AbstractTransformation{FixedVector{3}, FixedVector{3}}
+    angle::T
+    sin::T
+    cos::T
+end
+immutable RotationZX{T} <: AbstractTransformation{FixedVector{3}, FixedVector{3}}
+    angle::T
+    sin::T
+    cos::T
+end
+
+function RotationXY(a)
+    s = sin(a)
+    c = cos(a)
+    return RotationXY(a,s,c)
+end
+RotationYX(a) = RotationXY(-a)
+function RotationYZ(a)
+    s = sin(a)
+    c = cos(a)
+    return RotationYZ(a,s,c)
+end
+RotationZY(a) = RotationYZ(-a)
+function RotationZX(a)
+    s = sin(a)
+    c = cos(a)
+    return RotationZX(a,s,c)
+end
+RotationXZ(a) = RotationZX(-a)
+
+Base.show(io::IO, r::RotationXY) = print(io, "RotationXY($(r.angle))")
+Base.show(io::IO, r::RotationYZ) = print(io, "RotationYZ($(r.angle))")
+Base.show(io::IO, r::RotationZX) = print(io, "RotationZX($(r.angle))")
+
+function transform(trans::RotationXY, x::FixedVector{3})
+    (sincos, x2) = promote(Vec(trans.sin, trans.cos), x)
+    (typeof(x2))(x[1]*sincos[2] - x[2]*sincos[1], x[1]*sincos[1] + x[2]*sincos[2], x2[3])
```

Comment:
What I got is equivalent and can be faster (if the rotation angles are gradient numbers, for instance). 

What I should *really* do is find the promotion rule for `trans.sin * x[1]` and apply this to convert the the final term (`x2[3]` - > `convert(promote_type(trans.sin, x[1]), x[3])`)